desk/app/channels-server.hoon

```diff
@@ -836,4 +997,283 @@
       (said:utils nest plan posts.channel)
     (give %kick ~ ~)
   --
+++  scry-path
+  |=  [=dude:gall =path]
+  %+  welp
+  /(scot %p our.bowl)/[dude]/(scot %da now.bowl)
+  path
+++  get-hook-context
+  |=  [channel=(unit [nest:c v-channel:c]) =config:h]
+  ^-  context:h
+  =/  group
+    ?~  channel  ~
+    =*  flag  group.perm.perm.+.u.channel
+    %-  some
+    ?.  .^(? %gu (scry-path %groups /$))  *group-ui:g
+    ?.  .^(? %gx (scry-path %groups /exists/(scot %p p.flag)/[q.flag]/noun))
+      *group-ui:g
+    .^(group-ui:g %gx (scry-path %groups /groups/(scot %p p.flag)/[q.flag]/v1/noun))
+  :*  channel
+      group
+      v-channels
+      *hook:h  ::  we default this because each hook will replace with itself
+      config
+      [now our src eny]:bowl
+  ==
+::
+++  give-hook-response
+  |=  =response:h
+  ^+  cor
+  (give %fact ~[/hooks/v0] hooks-response-0+!>(response))
+++  ho-core
+  |_  [=id:h =hook:h gone=_|]
+  ++  ho-core  .
+  ++  emit  |=(=card ho-core(cor (^emit card)))
+  ++  emil  |=(caz=(list card) ho-core(cor (^emil caz)))
+  ++  give  |=(=gift:agent:gall ho-core(cor (^give gift)))
+  ++  ho-abet
+    %_  cor
+        hooks.hooks
+      ?:(gone (~(del by hooks.hooks) id) (~(put by hooks.hooks) id hook))
+    ==
+  ::
+  ++  ho-abed
+    |=  i=id:h
+    ho-core(id i, hook (~(got by hooks.hooks) i))
+  ::
+  ++  ho-add
+    |=  [name=@t src=@t]
+    ^+  ho-core
+    ~&  "adding hook {<name>}"
+    =.  id  (rsh [3 48] eny.bowl)
```

Comment:
128 bits is a lot (arguably too much for a local identifier), but doesn't _strictly_ guarantee uniqueness. You may want to loop inside the `=.` here until `!(~(has by hooks.hooks) id)`.

And, super stupid nitpick, but if you want consistent/maximum size here regardless of what happens to be put into `eny.bowl`, you want `+end` in place of `+rsh`.

```suggestion
    =.  id
      =+  i=(end 7 eny.bowl)
      |-(?:((~(has by hooks.hooks) i) $(i +(i)) i)
```